python/ray/train/torch.py

```diff
@@ -135,3 +140,178 @@ def on_shutdown(self, worker_group: WorkerGroup,
 
         worker_group.execute(
             shutdown_torch, destroy_process_group=len(worker_group) > 1)
+
+    @staticmethod
+    def encode_checkpoint(checkpoint: Dict) -> C:
+        """Special handling for moving model from worker to driver."""
+
+        # If model is being checkpointed and is wrapped in DDP, then extract
+        # out the underlying module. If not, then deserialization will fail
+        # since the torch process group is not initialized on the driver.
+
+        for k, v in checkpoint.items():
+            if isinstance(v, DistributedDataParallel) and hasattr(v, "module"):
+                checkpoint[k] = v.module
+
+        # Convert the checkpoint dict to bytes, so that any GPU tensors that
+        # are in the checkpoint dict can be properly deserialized on the
+        # driver side, even if the driver does not have access to a GPU device.
+        _buffer = io.BytesIO()
+        torch.save(checkpoint, _buffer)
+        return _buffer.getvalue()
```

Comment:
Nice, love the comments here! In the future it would be good to include this information in the public docs as well so users aren't confused (by how magical it is).